### PR TITLE
Upgrade kotlin-dsl-conventions plugin

### PR DESCRIPTION
**Pull request by wolfs**
To get rid of some deprecation warnings from the ktlint plugin:
https://ge.gradle.org/s/awrb5m62zpan2/deprecations

### CODE REVIEW

File: build-logic-commons/gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -25,6 +25,6 @@ dependencies {
     implementation("org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:4.3.0")
     // This Kotlin version should only be updated when updating the above kotlin-dsl version
     implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.22")
-    implementation("org.gradle.kotlin:gradle-kotlin-dsl-conventions:0.8.0")
+    implementation("org.gradle.kotlin:gradle-kotlin-dsl-conventions")
```
Comment (eskatos) [inline build-logic-commons/gradle-plugin/build.gradle.kts:28]: 👍

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -1403,9 +1408,9 @@
             <sha256 value="d645e5bf6fdd881541e085289e28fa298b64ac8d208a006ac649756074732a6f" origin="Verified" reason="Artifact is not signed"/>
          </artifact>
       </component>
-      <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-conventions" version="0.8.0">
-         <artifact name="gradle-kotlin-dsl-conventions-0.8.0.jar">
-            <sha256 value="b4182982109d85f9a623fa597932c3f09651467559bb99965c41d45159e89a68" origin="Verified" reason="Artifact is not signed"/>
+      <component group="org.gradle.kotlin" name="gradle-kotlin-dsl-conventions" version="0.9.0">
+         <artifact name="gradle-kotlin-dsl-conventions-0.9.0.jar">
+            <sha256 value="4cdec7506a466fb7c2d7579728a5b25e0d662a8f6c8291be31d3badaad55d922" origin="Generated by Gradle" reason="Artifact is not signed"/>
```
Comment (eskatos) [inline gradle/verification-metadata.xml:1413]: ❌ Should not be `Generated by Gradle`

File: gradle/verification-metadata.xml [hunk 3]
```diff
@@ -1463,9 +1468,9 @@
             <pgp value="073F7A9345756F3B40CDB99E6C70A3B7599C5736"/>
          </artifact>
       </component>
-      <component group="org.jlleitschuh.gradle" name="ktlint-gradle" version="10.3.0">
-         <artifact name="ktlint-gradle-10.3.0.jar">
-            <sha256 value="6923398b3a77a1fb3722aacbcdde6ab0527cfcc98eb3fef6974074f96ae7fe76" origin="Verified" reason="Artifact is not signed"/>
+      <component group="org.jlleitschuh.gradle" name="ktlint-gradle" version="12.1.0">
+         <artifact name="ktlint-gradle-12.1.0.jar">
+            <sha256 value="4ac232e64efb1e73674b3153c78b5decd476540b27cf21974e7e4919bae54188" origin="Generated by Gradle" reason="Artifact is not signed"/>
```
Comment (eskatos) [inline gradle/verification-metadata.xml:1473]: ❌ Should not be `Generated by Gradle`

**Review by eskatos — CHANGES_REQUESTED** [review]

File: gradle/verification-metadata.xml (hunk 2 above)
Comment (wolfs) [reply gradle/verification-metadata.xml]: Verified.

File: gradle/verification-metadata.xml (hunk 3 above)
Comment (wolfs) [reply gradle/verification-metadata.xml]: Verified.

**Review by eskatos — APPROVED** [review]

**Comment by wolfs** [comment]
@bot-gradle merge

**Pull request merged**